rules/store/kv/store.go

```diff
@@ -176,3 +177,11 @@ func (s *store) namespacesTransaction(nss *rules.Namespaces) (kv.Condition, kv.O
 
 	return namespacesCond, kv.NewSetOp(namespacesKey, nssSchema), nil
 }
+
+func wrapWriteError(err error) error {
+	if err == kv.ErrConditionCheckFailed {
+		return merrors.NewStaleDataError(err.Error())
```

Comment:
Can we perhaps provide more context around the operation that caused the error? I think this error will be propagated all the way back to the user so "condition check failed" is less clear than say "stale write request: condition check failed". Also does kv return which condition has failed? If so, might be good to include the failing condition in the error message too.

Reply:
I can take a look at what we get back from KV.

Reply:
I can update the error string but it looks like we dont get back the condition that failed right now. Ill see if that is something we can fix in the future.